src/checkoutservice/main.go

```diff
@@ -168,6 +168,58 @@ func (cs *checkoutService) Watch(req *healthpb.HealthCheckRequest, ws healthpb.H
 	return status.Errorf(codes.Unimplemented, "health check via Watch not implemented")
 }
 
+// Implemented GeneratePayment GRPC server API of checkoutService.
+// This API will return error in response.
+func (cs *checkoutService) GeneratePayment(ctx context.Context, req *pb.GeneratePaymentRequest) (*pb.Empty, error) {
+	// doSomeProcessing() is called for some processing delay in API call
+	doSomeProcessing()
+
+	// Return error
+	return new(pb.Empty), status.Errorf(codes.Internal, "Something went wrong with this request!")
+}
+
+// Implemented GenerateSalesTax GRPC server API of checkoutService.
+// This API will return error in response if country in request is france, else it will return success.
+func (cs *checkoutService) GenerateSalesTax(ctx context.Context, req *pb.GenerateSalesTaxRequest) (*pb.Empty, error) {
+	// doSomeProcessing() is called for some processing delay in API call
+	doSomeProcessing()
+
+	var country = "france"
+
+	// if country data provided in request is france then return error
+	// else return success
+	if req.Country == country {
+		time.Sleep(2 * time.Second)
+		if ctx.Err() == context.Canceled {
```

Comment:
It should be a select statement here like this:

```
		select {
		case <-ctx.Done():
			return fmt.Errorf("request is cancelled or timed out %w", err)
		case <-time.After(3 * time.Second):
		}
```

Reply:
Done